src/microbe_stage/editor/MicrobeEditorAction.cs

```diff
@@ -12,29 +12,29 @@ public class MicrobeEditorAction : ReversibleAction
     [JsonProperty]
     public readonly int Cost;
 
-    /// <summary>
-    ///   Action specific data
-    /// </summary>
     [JsonProperty]
-    public IMicrobeEditorActionData Data;
+    public readonly Action<MicrobeEditorAction> Redo;
 
     [JsonProperty]
-    private readonly Action<MicrobeEditorAction> redo;
+    public readonly Action<MicrobeEditorAction> Undo;
 
     [JsonProperty]
-    private readonly Action<MicrobeEditorAction> undo;
+    public readonly MicrobeEditor Editor;
 
+    /// <summary>
+    ///   Action specific data
+    /// </summary>
     [JsonProperty]
-    private readonly MicrobeEditor editor;
+    public IMicrobeEditorActionData Data;
```

Comment:
Could you restore the old order of those variables? This diff is very hard to read, at least for me.
And wouldn't properties with private setters be better here?

Reply:
Rider suggests readonly variables be placed before non-readonly ones. The change here is simply changing the modifiers from private to public.

Reply:
They are readonly ones. Can't be set outside constructor.

Reply:
Public *fields* should not be used. Use properties if needed, but I think there might be a better solution out there rather than doing this. I would rather add like a method to a rigidity change action class that allows modifying it, and then using a cast / type check to try to cast the latest action to that type to access the method.

Reply:
Okay, but I'm not fully sure what I've done is what you've said.

Reply:
These are *fields* not *properties* there is a big difference between those in C#.

Reply:
Actually I'm wondering if I've got *"add like a method to a rigidity change action class"* right. However, thanks for telling me not to use public fields.